Qast/Managers/FirebaseManager.swift

```diff
@@ -1,43 +1,50 @@
-import Foundation
-import Firebase
+//
+//  FirebaseManager.swift
+//  Qast
+//
+//  Created by Austin Welch on 7/5/19.
+//  Copyright © 2019 Qast. All rights reserved.
+//
+
 import FirebaseFirestore
 import CoreLocation.CLLocation
 
-let DB_BASE = Firestore.firestore()
-
-typealias SoundZoneCompletionBlock = ([SoundZone], Error?) -> Void
+struct FirebaseManager: SoundZoneAPI {
 
-class FirebaseManager {
-    static let instance = FirebaseManager()
```

Comment:
Without the static instance won't we end up with a different FirebaseManager instance doing the same work across many VC's? If we're not customizing the FirebaseManager for each VC's use case why not just keep it static?

Reply:
We can do either, the MapViewController is currently set up for dependency injection, which is a good route to take if we are doing a lot of testing and trying to avoid tight coupling. BUT using the singleton patter, similar to our SessionManager is also a valid approach. it really depends on what we want to do

Reply:
Unless each FirebaseManager needs to maintain its own state depending on which VC it's passed to, a static instance is a solid go to design principle-wise.

With repo pattern your protocol methods are generally stateless in regards to the class that implements them.

So for our current scale and use case deleting the instance isn't hurting but also isn't helping. And with it, we can define helpful constants used throughout, like the CollectionsReferences.